lucene/misc/src/java/org/apache/lucene/store/DirectIODirectory.java

```diff
@@ -164,15 +149,16 @@ public IndexOutput createOutput(String name, IOContext context) throws IOExcepti
     private long fileLength;
     private boolean isOpen;
 
-    public NativeUnixIndexOutput(Path path, String name, int bufferSize) throws IOException {
-      super("NativeUnixIndexOutput(path=\"" + path.toString() + "\")", name);
-      //this.path = path;
-      final FileDescriptor fd = NativePosixUtil.open_direct(path.toString(), false);
-      fos = new FileOutputStream(fd);
-      //fos = new FileOutputStream(path);
-      channel = fos.getChannel();
-      buffer = ByteBuffer.allocateDirect(bufferSize);
-      this.bufferSize = bufferSize;
+  @SuppressForbidden(reason = "com.sun.nio.file.ExtendedOpenOption: Direct I/O with FileChannel requires the use of internal proprietary API ExtendedOpenOption.DIRECT")
+  public DirectIOIndexOutput(Path path, String name) throws IOException {
+      super("DirectIOIndexOutput(path=\"" + path.toString() + "\")", name);
+
+      int blockSize = Math.toIntExact(Files.getFileStore(path).getBlockSize());
+      bufferSize = Math.addExact(blockSize, blockSize - 1);
+      buffer = ByteBuffer.allocateDirect(bufferSize).alignedSlice(blockSize);
```

Comment:
Hmm I wonder why we got away without using `.alignedSlice` before?

Reply:
`ByteBuffer.alignedSlice` was added since JDK 9 which was released in 2017, so I guess it's just the code hasn't been updated to take advantage of the new API?